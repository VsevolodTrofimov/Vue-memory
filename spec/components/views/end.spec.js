import Vuex from 'vuex'
import { mount, createLocalVue } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'

import EndVeiw from '@/src/components/views/End/End.vue'


const VSSR = createRenderer()
const localVue = createLocalVue()
localVue.config.productionTip = false
localVue.use(Vuex)

describe('Views', () => {
  describe('End View', () => {
    let actions
    let stubStore

    beforeEach(() => {
      actions = {
        startGame: jest.fn()
      }

      stubStore = new Vuex.Store({
        state: {},
        getters: {
          score: () => 10
        },
        actions
      })
    })

    it('matches snapshot', () => {
      const vm = new localVue({
        store: stubStore,
        render: h => h(EndVeiw)
      })

      VSSR.renderToString(vm, (err, str) => {
        expect(str).toMatchSnapshot()
      })
    })

    it('displays score from store', () => {
      const wrapper = mount(EndVeiw, {store: stubStore, localVue})
      const scoreNode = wrapper.find('[data-tid="EndGame-finalScore"]')
      expect(parseInt(scoreNode.text())).toEqual(stubStore.getters.score)
    })

    it('dispatches startGame action', () => {
      const wrapper = mount(EndVeiw, {store: stubStore, localVue})
      wrapper.find('[data-tid="EndGame-retryGame"]').trigger('click')
      expect(actions.startGame).toBeCalled()
    })
  })
})