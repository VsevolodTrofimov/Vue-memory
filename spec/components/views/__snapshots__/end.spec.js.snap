// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Views End View matches snapshot 1`] = `
<div class="view">
    <img srcset="/static/banners/EndGame.png, /static/banners/EndGame@2x.png 2x" alt="Flipped cards" class="banner">
    <h2 class="heading title">
        Поздравляем!
        <br> Ваш итоговый счет: <span data-tid="EndGame-finalScore">10</span></h2>
    <button data-tid="EndGame-retryGame" class="btn"> Еще раз </button>
</div>
`;
